pytorch_lightning/trainer/trainer.py

```diff
@@ -1072,13 +1093,16 @@ def _run_sanity_check(self, ref_model):
             # restore the previous stage when the sanity check if finished
             self.state.stage = stage
 
-    def __load_ckpt_weights(self, ckpt_path: Optional[str]) -> Optional[str]:
-        if ckpt_path is None:
-            return
+    def __set_ckpt_path(self, ckpt_path: Optional[str], model_provided: bool, model_connected: bool) -> Optional[str]:
+        if model_provided and (ckpt_path in ('best', None)):
+            return  # use passed model to function without loading weights
+
+        if model_connected and ckpt_path is None:
+            return  # use connected model without loading weights
```

Comment:
Move this condition to the top so we return if `ckpt_path` is None? (as we had in the old code)